common/lib/xmodule/xmodule/js/src/capa/display.js

```diff
@@ -1157,58 +1154,64 @@
+                return that.enableButtons(initiallyEnabledButtons, true, isFromCheckOperation);
             });
         };
 
         /**
-         * Used to enable/disable all buttons in problem.
+         * Enables all buttons by removing the disabled attribute. The submit button is checked separately due to
+         *     the changing text it contains.
          *
          * params:
-         *     'enable' is a boolean to determine enabling/disabling of buttons.
-         *     'isFromCheckOperation' is a boolean to keep track if operation was initiated
+         *     'buttons' is an array of buttons that will have their 'disabled' attribute deleted
+         *     'enable' a boolean to either enable or disable the buttons passed in the first parameter
+         *     'changeSubmitButtonText' is a boolean to keep track if operation was initiated
          *         from submit so that text of submit button will also be changed while disabling/enabling
          *         the submit button.
          */
-        Problem.prototype.enableAllButtons = function(enable, isFromCheckOperation) {
-            // Called by disableAllButtonsWhileRunning to automatically disable all buttons while check,reset, or
-            // save internal are running. Then enable all the buttons again after it is done.
-            if (enable) {
-                this.resetButton.add(this.saveButton).add(this.hintButton).add(this.showButton).
-                    removeAttr('disabled');
-            } else {
-                this.resetButton.add(this.saveButton).add(this.hintButton).add(this.showButton).
-                    attr({disabled: 'disabled'});
-            }
-            return this.enableSubmitButton(enable, isFromCheckOperation);
+        Problem.prototype.enableButtons = function(buttons, enable, changeSubmitButtonText) {
+            var that = this;
+            buttons.forEach(function(button) {
+                if (button === that.submitButton) {
+                    that.enableSubmitButton(enable, changeSubmitButtonText);
+                } else if (enable) {
+                    button.removeAttr('disabled');
+                } else {
+                    button.attr({disabled: 'disabled'});
+                }
+            });
         };
 
         /**
          *  Used to disable submit button to reduce chance of accidental double-submissions.
          *
          * params:
          *     'enable' is a boolean to determine enabling/disabling of submit button.
-         *     'changeText' is a boolean to determine if there is need to change the
+         *     'changeSubmitButtonText' is a boolean to determine if there is need to change the
          *         text of submit button as well.
          */
-        Problem.prototype.enableSubmitButton = function(enable, changeText) {
+        Problem.prototype.enableSubmitButton = function(enable, changeSubmitButtonText) {
```

Comment:
Nit: you could keep this just "changeText" in this method because the method itself is scoped to the Submit button.